src/Page.js

```diff
@@ -55,48 +55,212 @@ cheerio.prototype.options.xmlMode = true; // Enable xml mode for self-closing ta
 cheerio.prototype.options.decodeEntities = false; // Don't escape HTML entities
 
 class Page {
+  /**
+   * A page configuration object.
+   * @typedef {Object<string, any>} PageConfig
+   * @property {Object<string, any>} asset
+   * @property {string} baseUrl
+   * @property {Set<any>} baseUrlMap
+   * @property {string} content
+   * @property {string} faviconUrl
+   * @property {Object<string, any>} frontmatter
+   * @property {string} layout
+   * @property {string} layoutsAssetPath
+   * @property {string} rootPath
+   * @property {boolean} enableSearch
+   * @property {boolean} globalOverride
+   * @property {Array} plugins
```

Comment:
Should we label the type of the content in the array?

Reply:
The content in `plugin` array is the `require`d plugin objects (e.g. `plugins['name']=require('name');`)